extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java

```diff
@@ -130,4 +130,16 @@
      */
     @ConfigItem
     public Optional<String> validationQuerySql;
+
+    /**
+     * When enabled jdbc url will be forged to enable google socket factory
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean useGcp;
```

Comment:
Just wondering, would it be possible to automate this in any way?

Would be great if we could detect it's being run on GCP and not need such a configuration flag

Reply:
we can detect we are running on GCP by connecting to the `metadata.google.internal` as followed : 
```java 
try {
        Socket socket = new Socket("meta.google.internal", 80);
        System.out.println("Running in GCP!!");
    } catch (IOException e) {
        e.printStackTrace();
    }
```

But as stated in the [Deploying to google cloud documentation](docs/src/main/asciidoc/deploying-to-google-cloud.adoc), connecting with service account is just an option. I don't think Quarkus should force it. We could combine this detection and the presence of the cloudSqlInstance property to automate this.

Again as stated in the https://github.com/quarkusio/quarkus/issues/5824, for it to work as is, we would need to add a dependency to `com.google.cloud.sql:postgres-socket-factory:1.0.15` and/or `com.google.cloud.sql:mysql-socket-factory-connector-j-8:1.0.15` plus dependencies to the corresponding drivers.

As of now it is not possible as maven-enforcer-plugin will fail the build.

Alternatively, we could delegate the responsibility to the applications using the quarkus-agroal extension.

Reply:
I have removed postgres and postgres-socket-factory so that the build doesn't fail.

Again we could force the use of service account, but it would be making decisions instead of letting developper choose their connection mode